Content.Server/TapeRecorder/TapeRecorderSystem.cs

```diff
@@ -0,0 +1,344 @@
+using System.Linq;
+using Content.Server.Chat;
+using Content.Server.Chat.Systems;
+using Content.Server.Popups;
+using Content.Shared.Chat;
+using Content.Shared.Containers.ItemSlots;
+using Content.Shared.Examine;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Verbs;
+using Content.Shared.TapeRecorder;
+using Robust.Shared.Player;
+using Robust.Server.Containers;
+using Robust.Shared.Containers;
+
+namespace Content.Server.TapeRecorder
+{
+    /// <summary>
+    /// System for handling tape recorders
+    /// </summary>
+    public sealed class TapeRecorderSystem : EntitySystem
+    {
+        [Dependency] private readonly ChatSystem _chat = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+        [Dependency] private readonly ContainerSystem _containerSystem = default!;
+        [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<TapeRecorderComponent, ChatMessageHeardNearbyEvent>(OnChatMessageHeard);
+            SubscribeLocalEvent<TapeRecorderComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<TapeRecorderComponent, GetVerbsEvent<AlternativeVerb>>(OnGetAltVerbs);
+            SubscribeLocalEvent<TapeRecorderComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<TapeRecorderComponent, ItemSlotChangedEvent>(OnItemSlotChanged);
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var (tapeRecorder, activeTapeRecorder) in EntityManager.EntityQuery<TapeRecorderComponent, ActiveTapeRecorderComponent>())
```

Comment:
If going with the accumulated time solution for this, then maybe it wouldn't be wise to do an entity query that looks like this, otherwise your tape recorders won't get the accumulated time they're tracking.